src/utils/serializers/arrays.ts

```diff
@@ -39,3 +69,174 @@ export function deserializeObj<T extends ISerializable<T>>(
+      case TypedArrayUnit.I32:
+        view.setInt32(0, value, true);
+        break;
+      case TypedArrayUnit.I64:
+        view.setBigInt64(0, BigInt(value), true);
+        break;
+      case TypedArrayUnit.U32:
+        view.setUint32(0, value, true);
+        break;
+      case TypedArrayUnit.U64:
+        view.setBigUint64(0, BigInt(value), true);
+        break;
+      default:
+        throw new Error(`Unsupported type ${type}`);
+    }
+  }
+  return target;
+}
+
+/**
+ * Converts a StaticArray<u8> into an Array of type parameter.
+ *
+ * @remarks
+ * inspired by https://github.com/AssemblyScript/assemblyscript/blob/main/std/assembly/array.ts#L69-L81
+ *
+ * @param source - the array to convert
+ */
+export function bytesToNativeTypeArray<T extends TypedArrayUnit>(
+  source: Uint8Array,
+  typedArrayType: T,
+): T[] {
+  const sourceLength = source.length;
+  // Calculate the total length of the array
+  let targetLength = sourceLength / getDatatypeSize(typedArrayType);
+  if (!(targetLength % 1 === 0)) {
+    throw new Error(`None-integer array length computation`);
+  }
+  const dataView = new DataView(source.buffer);
+
+  // Create a new typed array of type T and fill it with the converted values
+  const result: T[] = [];
+  let byteOffset = 0;
+  for (let i = 0; i < targetLength; i++) {
+    const size = getDatatypeSize(typedArrayType);
+    switch (typedArrayType) {
+      case TypedArrayUnit.STRING:
+        let dataArr = [];
+        for (let j = 0; j < size; j++) {
+          const letter = dataView.getUint8(byteOffset + j);
+          if (letter <= 0) continue;
+          dataArr.push(letter);
+        }
+        result.push(bytesToStr(Uint8Array.from(dataArr)) as unknown as T);
+        byteOffset += size;
+        break;
+      case TypedArrayUnit.BOOL:
+        const boolVal = source[i] === 1 ? true : false;
+        result.push(boolVal as unknown as T);
+        byteOffset += size;
+        break;
+      case TypedArrayUnit.U8:
+        result.push(source[i] as unknown as T);
+        byteOffset += size;
+        break;
+      case TypedArrayUnit.F32:
+        result.push(dataView.getFloat32(byteOffset, true) as unknown as T);
+        byteOffset += size;
+        break;
+      case TypedArrayUnit.F64:
+        result.push(dataView.getFloat64(byteOffset, true) as unknown as T);
+        byteOffset += size;
+        break;
+      case TypedArrayUnit.I32:
+        result.push(dataView.getInt32(byteOffset, true) as unknown as T);
+        byteOffset += size;
+        break;
+      case TypedArrayUnit.I64:
+        result.push(dataView.getBigInt64(byteOffset, true) as unknown as T);
+        byteOffset += size;
+        break;
+      case TypedArrayUnit.U32:
+        result.push(dataView.getUint32(byteOffset, true) as unknown as T);
+        byteOffset += size;
+        break;
+      case TypedArrayUnit.U64:
+        result.push(dataView.getBigUint64(byteOffset, true) as unknown as T);
+        byteOffset += size;
```

Comment:
this instruction is executed in every cases of the switch, consider doing it one time just after the switch

Reply:
> this instruction is executed in every cases of the switch, consider doing it one time just after the switch

Thanks, fixed that. Good point!